operation/fhir-operation-versions/src/main/java/com/ibm/fhir/operation/versions/VersionsOperation.java

```diff
@@ -0,0 +1,213 @@
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+import com.ibm.fhir.config.FHIRConfigHelper;
+import com.ibm.fhir.config.FHIRConfiguration;
+import com.ibm.fhir.core.FHIRVersionParam;
+import com.ibm.fhir.exception.FHIROperationException;
+import com.ibm.fhir.model.resource.OperationDefinition;
+import com.ibm.fhir.model.resource.Parameters;
+import com.ibm.fhir.model.resource.Parameters.Parameter;
+import com.ibm.fhir.model.resource.Resource;
+import com.ibm.fhir.model.type.Code;
+import com.ibm.fhir.registry.FHIRRegistry;
+import com.ibm.fhir.server.spi.operation.AbstractOperation;
+import com.ibm.fhir.server.spi.operation.FHIROperationContext;
+import com.ibm.fhir.server.spi.operation.FHIROperationUtil;
+import com.ibm.fhir.server.spi.operation.FHIRResourceHelpers;
+
+import jakarta.json.Json;
+import jakarta.json.JsonArrayBuilder;
+import jakarta.json.JsonBuilderFactory;
+import jakarta.json.JsonObject;
+
+/**
+ * Implementation of the $versions operation, which returns the list of supported FHIR versions,
+ * along with the default FHIR version.
+ */
+public class VersionsOperation extends AbstractOperation {
+
+    private static final JsonBuilderFactory JSON_BUILDER_FACTORY = Json.createBuilderFactory(null);
+    private static final DocumentBuilderFactory XML_DOM_FACTORY = DocumentBuilderFactory.newInstance();
+    private static final String ACCEPT_HEADER = "Accept";
+    private static final String PARAM_DEFAULT = "default";
+    private static final String PARAM_VERSION = "version";
+    private static final String PARAM_VERSIONS = "versions";
+    private static final int STATUS_OK = 200;
+
+    public VersionsOperation() {
+        super();
+    }
+
+    @Override
+    protected OperationDefinition buildOperationDefinition() {
+        return FHIRRegistry.getInstance().getResource("http://hl7.org/fhir/OperationDefinition/CapabilityStatement-versions",
+                OperationDefinition.class);
+    }
+
+    @Override
+    protected Parameters doInvoke(FHIROperationContext operationContext, Class<? extends Resource> resourceType,
+            String logicalId, String versionId, Parameters parameters, FHIRResourceHelpers resourceHelper)
+            throws FHIROperationException {
+        try {
+            List<String> versions = getFhirVersions();
+            String defaultVersion = getDefaultFhirVersion();
+
+            // Build and return response based on the Accept header
+            String outputFormat = getNonFhirOutputFormat(operationContext);
+            if (MediaType.APPLICATION_JSON.equals(outputFormat)) {
+                return buildJsonResponse(operationContext, versions, defaultVersion);
+            } else if (MediaType.APPLICATION_XML.equals(outputFormat)) {
+                return buildXmlResponse(operationContext, versions, defaultVersion);
+            } else {
+                return buildFhirResponse(versions, defaultVersion);
+            }
```

Comment:
Why not defer to FHIRProvider, it'll transpose from the model to the output format for the version based on the Accept Header? Then the framework will change it. 

Is this due to a non-standard output?

Reply:
I added a README.md to explain the various response formats.